tests/test_google_ads_interrupted_sync_add_stream.py

```diff
@@ -0,0 +1,269 @@
+import re
+import os
+from datetime import datetime as dt
+from datetime import timedelta
+
+from tap_tester import menagerie, connections, runner
+
+from base import GoogleAdsBase
+
+
+class InterruptedSyncAddStreamTest(GoogleAdsBase):
+    """Test tap's ability to recover from an interrupted sync"""
+
+    @staticmethod
+    def name():
+        return "tt_google_ads_interruption_add"
+
+    def get_properties(self, original: bool = True):
+        """Configurable properties, with a switch to override the 'start_date' property"""
+        return_value = {
+            'start_date':   '2021-12-01T00:00:00Z',
+            'user_id':      'not used?', # TODO ?
+            'conversion_window': '1',  # days
+            'customer_ids': ','.join(self.get_customer_ids()),
+            'login_customer_ids': [{"customerId": os.getenv('TAP_GOOGLE_ADS_CUSTOMER_ID'),
+                                    "loginCustomerId": os.getenv('TAP_GOOGLE_ADS_LOGIN_CUSTOMER_ID'),}],
+
+        }
+
+        # TODO_TDL-17911 Add a test around conversion_window_days
+        if original:
+            return return_value
+
+        return_value["start_date"] = self.start_date
+        return return_value
+
+
+    def assertIsDateFormat(self, value, str_format): # TODO needed in this test?
```

Comment:
This can be moved to bottom of base.py under Tap Specific methods and the definition can be removed from tests where it is used.